src/test/kotlin/se/uulm/snowballr/backend/service/project/GetProjectInformationTest.kt

```diff
@@ -0,0 +1,94 @@
+package se.uulm.snowballr.backend.service.project
+
+import com.google.protobuf.util.FieldMaskUtil
+import io.mockk.coEvery
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import se.uulm.snowballr.backend.DataBuilder
+import se.uulm.snowballr.backend.model.SnowballRException.NotFoundException
+import se.uulm.snowballr.backend.model.SnowballRException.UnauthorizedException
+import se.uulm.snowballr.backend.service.MainServiceTest
+import snowballr.ProjectOuterClass
+import java.time.OffsetDateTime
+import java.util.UUID
+
+class GetProjectInformationTest : MainServiceTest() {
+    private fun getRequest(projectId: UUID, paths: List<String>? = null) = ProjectOuterClass.Project.Information.Get
+        .newBuilder()
+        .setProjectId(projectId.toString())
+        .also { if (paths != null) it.setMask(FieldMaskUtil.fromStringList(paths)) }
+        .build()
+
+    @Test
+    fun `When the project does not exist, then a NotFoundException is thrown`() = runTest {
+        val user = DataBuilder.createExampleUser()
+        val projectId = UUID.randomUUID()
+
+        mockCurrentUser(user)
+        coEvery { projectRepoMock.doesProjectExistById(projectId) } returns false
+
+        assertThrows<NotFoundException> { mainService.getProjectInformation(getRequest(projectId)) }
+    }
+
+    @Test
+    fun `When the user is not a member of the project, then an UnauthorizedException is thrown`() = runTest {
+        val user = DataBuilder.createExampleUser()
+        val project = DataBuilder.createExampleProject()
+
+        mockCurrentUser(user)
+        coEvery { projectMemberRepoMock.getProjectMembers(project.id) } returns emptyList()
+        coEvery { projectRepoMock.doesProjectExistById(project.id) } returns true
+
+        assertThrows<UnauthorizedException.Single> { mainService.getProjectInformation(getRequest(project.id)) }
```

Comment:
We don't care about the `Single` or `All` subtype in this case.
```suggestion
        assertThrows<UnauthorizedException> { mainService.getProjectInformation(getRequest(project.id)) }
```